server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -399,30 +398,23 @@ public DocumentMapperForType documentMapperWithAutoCreate() {
      * Given the full name of a field, returns its {@link MappedFieldType}.
      */
     public MappedFieldType fieldType(String fullName) {
-        if (fullName.equals(TypeFieldType.NAME)) {
-            return new TypeFieldType(this.mapper == null ? "_doc" : this.mapper.type());
-        }
-        return this.mapper == null ? null : this.mapper.mappers().fieldTypes().get(fullName);
```

Comment:
let's track this as a follow-up so we can keep this change as contained as possible?

Reply:
Given how folks received my last attempt at this I wanted to keep my change as small as I could manage. I think it is correct to switch them all to `mappingLookup` and my instinct is to do it now, while I'm here. But I don't want to make this harder to review.

Reply:
I think that would not only make it harder to review, but it would make the PR bigger by making changes that are not strictly necessary. I think we should totally do it, but as a follow-up which is not going to be backported to 7.11

Reply:
It works for me to do it in a follow-up.